src/scales/scale.linear.js

```diff
@@ -33,16 +33,16 @@ module.exports = function(Chart) {
 
 				helpers.each(datasets, function(dataset, datasetIndex) {
 					var meta = chart.getDatasetMeta(datasetIndex);
-					if (valuesPerType[meta.type] === undefined) {
-						valuesPerType[meta.type] = {
+					if (valuesPerType[meta.type + meta.stack] === undefined) {
```

Comment:
Hm... its somewhat similar to meta.type in that it's just another property to group datasets. Maybe a common meta property that factors in both type and (an optional stack) would be better? Might be too big a change though.

Can't think of anything nicer :-(